erc20-migration/contracts/ZTESTBackupVault.sol

```diff
@@ -35,27 +35,34 @@ contract ZTESTBackupVault is Ownable {
     error AddressNotValid();
     error CumulativeHashNotValid();
     error CumulativeHashCheckpointReached();
+    error CumulativeHashCehckpointNotSet();
     error UnauthorizedOperation();
     error ERC20NotSet();
     error NothingToDistribute();
 
 
     /// @notice Smart contract constructor
     /// @param _admin  the only entity authorized to performe restore and distribution operations
+    constructor(address _admin) Ownable(_admin) {   
+        _cumulativeHash = bytes32(0);        
+        nextRewardIndex = 0;
+    }
+
+    /// @notice Set expected cumulative hash after all the data has been loaded
     /// @param _cumulativeHashCheckpoint  a cumulative recursive  hash calculated with all the dump data.
     ///                                   Will be used to verify the consistency of the restored data, and as
     ///                                   a checkpoint to understand when all the data has been loaded and the distribution 
     ///                                   can start
-    constructor(address _admin, bytes32 _cumulativeHashCheckpoint) Ownable(_admin) {
-        if(_cumulativeHashCheckpoint == bytes32(0)) revert CumulativeHashNotValid();   
-        _cumulativeHash = bytes32(0);
+    function setCumulativeHashCeckpoint(bytes32 _cumulativeHashCheckpoint) public onlyOwner{
+        if(_cumulativeHashCheckpoint == bytes32(0)) revert CumulativeHashNotValid();  
+        if (cumulativeHashCheckpoint != bytes32(0)) revert UnauthorizedOperation();  //already set
```

Comment:
I suppose cumulativeHashCheckpoint is implicitly initialized as bytes32(0), but to be consistent with _cumulativeHash maybe it is better to intialize it in the costructor.
In addition, in the `distribute` function it should be checked if cumulativeHashCheckpoint has been initialized. It is true that it should revert for NothingToDistribute, but actually the problem is that the initialization is missing.

Reply:
For the init part: it's consistent since both are not intialized explicitly (bytes32(0) is already the default value)